pennylane/tensorbox/fn.py

```diff
@@ -0,0 +1,416 @@
+    >>> cast(x, np.float64)
+    tensor([1., 2.], dtype=torch.float64)
+
+    We can also use strings:
+
+    >>> x = tf.Variable([1, 2])
+    >>> cast(x, "complex128")
+    <tf.Tensor: shape=(2,), dtype=complex128, numpy=array([1.+0.j, 2.+0.j])>
+    """
+    return TensorBox(tensor).cast(dtype).data
+
+
+def cast_like(tensor1, tensor2):
+    """Casts a tensor to the same dtype as another.
+
+    Args:
+        tensor1 (tensor_like): tensor to cast
+        tensor2 (tensor_like): tensor with corresponding dtype to cast to
+
+    Returns:
+        tensor_like: a tensor with the same shape and values as ``tensor1`` and the
+        same dtype as ``tensor2``.
+
+    **Example**
+
+    >>> x = torch.tensor([1, 2])
+    >>> y = torch.tensor([3., 4.])
+    >>> cast(x, y)
+    tensor([1., 2.])
+    """
+    dtype = TensorBox(tensor2).numpy().dtype.type
+    return TensorBox(tensor1).cast(dtype).data
+
+
+def convert_like(tensor1, tensor2):
+    """Convert a tensor to the same type as another.
+
+    Args:
+        tensor1 (tensor_like): tensor to convert
+        tensor2 (tensor_like): tensor with corresponding type to convert to
+
+    Returns:
+        tensor_like: a tensor with the same shape, values, and dtype as ``tensor1`` and the
+        same type as ``tensor2``.
+
+    **Example**
+
+    >>> x = np.array([1, 2])
+    >>> y = tf.Variable([3, 4])
+    >>> cast(x, y)
+    <tf.Tensor: shape=(2,), dtype=int64, numpy=array([1, 2])>
+    """
+    return TensorBox(tensor2).astensor(tensor1)
+
+
+def expand_dims(tensor, axis):
+    """Expand the shape of an array by adding a new dimensions of size 1
+    at the specified axis location.
+
+    .. warning::
+
+        This function differs from ``np.expand_dims``
+
+    Args:
+        tensor (tensor_like): tensor to expand
+        axis (int): location in the axes to place the new dimension
+
+    Returns:
+        tensor_like: a tensor with the expanded shape
+
+    **Example**
+
+    >>> x = tf.Variable([3, 4])
+    >>> expand_dims(x, axis=1)
+    <tf.Tensor: shape=(2, 1), dtype=int32, numpy=
+    array([[3],
+           [4]], dtype=int32)>
+    """
+    return TensorBox(tensor).expand_dims(axis).data
+
+
+def get_interface(tensor):
+    """Returns the name of the package that any array/tensor manipulations
+    will dispatch to. The returned strings correspond to those used for PennyLane
+    :doc:`interfaces </introduction/interfaces>`.
+
+    Args:
+        tensor (tensor_like): tensor input
+
+    Returns:
+        str: name of the interface
+
+    **Example**
+
+    >>> x = torch.tensor([1., 2.])
+    >>> get_interface(x)
+    'torch'
+    >>> from pennylane import numpy as np
+    >>> x = np.array([4, 5], requires_grad=True)
+    >>> get_interface(x)
+    'autograd'
+    """
+    return TensorBox(tensor).interface
+
+
+def toarray(tensor):
+    """Returns the tensor as a NumPy ``ndarray``. No copying
+    is performed; the tensor and the returned array share the
+    same storage.

```

Comment:
are you referring to 'in-place' manipulation?

Reply:
if the user calls this function without catching the returned object, will tensor be overwritten?

Reply:
No not quite, more so that the underlying memory representing the array data is the same.

This is how numpy/torch/tf tend to work a lot under the hood (see https://numpy.org/doc/stable/reference/generated/numpy.ndarray.view.html)